src/main/java/ca/bc/gov/backendstartapi/dto/UserDto.java

```diff
@@ -0,0 +1,56 @@
+package ca.bc.gov.backendstartapi.dto;
+
+import ca.bc.gov.backendstartapi.response.BaseResponse;
+import ca.bc.gov.backendstartapi.util.Empty;
+import java.util.Objects;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+/** This class represents a User data transition object. */
+@Builder
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+public class UserDto implements BaseResponse, Empty {
+
+  @Size(min = 2, max = 20)
+  @NotNull
+  private String firstName;
+
+  @Size(min = 2, max = 20)
+  @NotNull
+  private String lastName;
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    UserDto userDto = (UserDto) o;
+    return firstName.equals(userDto.firstName) && lastName.equals(userDto.lastName);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(firstName, lastName);
+  }
+
+  @Override
+  public String toString() {
+    return "UserDto{" + "firstName='" + firstName + '\'' + ", lastName='" + lastName + '\'' + '}';
+  }
+
+  @Override
+  public boolean isEmpty() {
+    return hashCode() == 961;
```

Comment:
Because that's the hash value when the instance (all fields) is empty, also considering null and blank spaces. Using hash code values is a good idea since JVM guarantee the idempotent result.